### PR TITLE
Use Benchmark-ips for Rails benchmarks.

### PR DESCRIPTION
**Pull request by tgxworld**
@SamSaffron @kirs I updated the Rails benchmarks to use `Benchmark-ips`. Thoughts?

### CODE REVIEW

**Comment by SamSaffron** [comment]
This is nice less duplicate code! 

We should amend our helper to measure how many objects are allocated for the block, which would be awesome and I would recommend adding. Number of objects allocated is a great measure.

**Comment by SamSaffron** [comment]
To get the count its quite simple..

after ips runs 
1. grab GC.stat[:total_allocated_object]
2. run a single block
3. grab GC.stat[:total_allocated_object]
4. add to json

Keep in mind the name changed here across some ruby versions so account for it.

**Comment by kirs** [comment]
:+1:

File: rails/benchmarks/support/benchmark_rails.rb [hunk 1]
```diff
@@ -11,40 +14,39 @@ def rails(n=1000, label=nil, disable_gc: true, warmup: 3)
...
+    def get_total_allocated_objects
+      if block_given?
+        key =
+          if RUBY_VERSION < '2.2'
+            :total_allocated_object
+          else
+            :total_allocated_objects
+          end
+
+        GC.start
```
Comment (tgxworld) [inline rails/benchmarks/support/benchmark_rails.rb]: Running GC before we count the `total_allocated_objects`/

**Comment by tgxworld** [comment]
@SamSaffron I've included `total_allocated_objects` in the output. :smile:

**Comment by SamSaffron** [comment]
not sure why you are running a GC there, what is the reasoning?

**Comment by tgxworld** [comment]
@SamSaffron Ah ok. While it doesn't matter for `disable_gc: true`, I think we need to run the GC once for `disable_gc: false`. This is to ensure that we don't end up running GC due to the previous allocations from the benchmark. Let me know if that makes sense? :smile_cat:

**Comment by tgxworld** [comment]
@SamSaffron Ping :smiley_cat:

**Comment by SamSaffron** [comment]
sorry, yeah you can merge this, but really no need for gc collect between
the time you start and stop measuring how many object a block allocated.

The counter in GC.stat[:total_allocated_object] is ever increasing, it
never goes done during a processes lifetime

**Pull request merged**